File: src/main/java/com/mbayou/kick4k/livestreams/LivestreamsClient.java
package com.mbayou.kick4k.livestreams;

import com.fasterxml.jackson.core.type.TypeReference;
import com.fasterxml.jackson.databind.ObjectMapper;
import com.mbayou.kick4k.KickConfiguration;
import com.mbayou.kick4k.api.ApiClient;
import com.mbayou.kick4k.authorization.AuthorizationClient;

import java.net.http.HttpClient;
import java.util.List;

public class LivestreamsClient extends ApiClient {
    public LivestreamsClient(HttpClient httpClient, ObjectMapper mapper, KickConfiguration configuration, AuthorizationClient authorization) {
        super(httpClient, mapper, configuration, authorization);
    }

    public Livestream getLivestream(Integer broadcasterUserId) {
        return this.getLivestreams(GetLivestreamsRequest.builder()
                .broadcasterUserId(broadcasterUserId).build())
                .stream()
                .findFirst()
                .orElse(null);
    }

    public List<Livestream> getLivestreams(GetLivestreamsRequest request) {
        return this.get(this.configuration.getLivestreams())
                .queryParams(request)
                .send(new TypeReference<>() {});
    }

    public LivestreamsStats getLivestreamsStats() {
        return this.get(this.configuration.getLivestreamsStats())
                .send(new TypeReference<>() {});
    }
}

File: src/main/kotlin/com/mbayou/kick4k/livestreams/LivestreamsClient.kt
package com.mbayou.kick4k.livestreams

import com.fasterxml.jackson.databind.ObjectMapper
import com.mbayou.kick4k.KickConfiguration
import com.mbayou.kick4k.api.ApiClient
import java.net.http.HttpClient

class LivestreamsClient(
    httpClient: HttpClient,
    mapper: ObjectMapper,
    configuration: KickConfiguration,
): ApiClient(httpClient, mapper, configuration) {

    fun getLivestream(accessToken: String, broadcasterUserId: Int): Livestream? {
        return getLivestreams(accessToken, GetLivestreamsRequest.builder().broadcasterUserId(broadcasterUserId).build()).firstOrNull()
    }

    fun getLivestreams(accessToken: String, request: GetLivestreamsRequest): List<Livestream> {
        return get(configuration.livestreams)
            .queryParams(request)
            .withAccessToken(accessToken)
            .send(responseType<List<Livestream>>()) ?: emptyList()
    }

    fun getLivestreamsStats(accessToken: String): LivestreamsStats? {
        return get(configuration.livestreamsStats)
            .withAccessToken(accessToken)
            .send(responseType<LivestreamsStats>())
    }
}
